lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,283 @@
+      });
+
+      expect(mockOnClose).toHaveBeenCalled();
+    });
+
+    it('executes onClose ONLY when pressing ESC key', async function () {
+      const user = userEvent.setup();
+
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const drawer = getDrawer({ onClose: mockOnClose })!;
+
+      await user.keyboard('Tab');
+      await user.keyboard('Enter');
+      await user.keyboard('q');
+      await user.keyboard('Q');
+
+      fireEvent.animationEnd(drawer);
+
+      expect(mockOnClose).not.toHaveBeenCalled();
+    });
+
+    it('executes onCancel when drawer is closed and animation is completed', async function() {
+      const user = userEvent.setup();
+      const mockOnCancel = jest.fn();
+
+      const Fixture = () => {
+        const [open, toggleDrawer] = useToggle(true);
+        return (
+          <NxDrawer open={open} onClose={toggleDrawer} onCancel={mockOnCancel}>
+            <NxDrawer.Header>
+              <NxDrawer.HeaderTitle>Title</NxDrawer.HeaderTitle>
+            </NxDrawer.Header>
+          </NxDrawer>
+        );
+      };
+
+      render(<Fixture />);
+
+      const closeButton = screen.getByRole('button', { name: 'Close', hidden: true });
+      const dialog = screen.getByRole('dialog', { hidden: true });
+
+      await act(async () => {
+        await user.click(closeButton);
+        await fireEvent.animationEnd(dialog);
+      });
+
+      await waitFor(() => expect(mockOnCancel).toHaveBeenCalled());
+    });
+
+    it('executes onClose when clicked outside of the drawer', async function() {
+      const documentAddEventListener = document.addEventListener;
+
+      try {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const eventMap: any = {};
+
+        document.addEventListener = jest.fn((eventType: string, callback: () => void) => {
+          eventMap[eventType] = callback;
+        }) as jest.Mock;
+
+        const mockOnClose = jest.fn();
+
+        render(
+          <div>
+            <NxButton>Outside</NxButton>
+            <NxDrawer open={true} onClose={mockOnClose}>
+              <NxButton>Inside</NxButton>
+            </NxDrawer>
+          </div>
+        );
+
+        const outsideButton = screen.getByRole('button', { name: 'Outside' });
+        const insideButton = screen.getByRole('button', { name: 'Inside', hidden: true });
+
+        expect(mockOnClose).not.toHaveBeenCalled();
+
+        await act(async () => {
+          await eventMap.click({ target: insideButton });
+        });
+
+        expect(mockOnClose).not.toHaveBeenCalled();
+
+        await act(async () => {
+          await eventMap.click({ target: outsideButton });
+        });
+
+        expect(mockOnClose).toHaveBeenCalled();
+      }
+      finally {
+        document.addEventListener = documentAddEventListener;
+      }
+    });
+  });
+
+  describe('NxDrawer Header', function() {
```

Comment:
Describe block names get concatenated together, so we really don't want `NxDrawer` repeated here, it's redundant with the parent describe block.

Reply:
Fixed here:

> Describe block names get concatenated together, so we really don't want `NxDrawer` repeated here, it's redundant with the parent describe block.

Fixed: 513ecbb4e57aae15bf56db3d8d5bcf38e3b91ed0